exetera/core/fields.py

```diff
@@ -303,27 +328,13 @@ def dtype(self):
         """
         return self._dataset.dtype
 
-    def register_reference(self, field: Field):
-        """
-
-        """
-        self._references.append(field)
-
-    def detach_reference(self, field: Field):
-        """
-
-        """
-        self._references.remove(field)
-
-    def concreate_all_fields(self):
-        """
-
-        """
-        for field in self._references:
-            field.concrete_reference()
-
     def __getitem__(self, item):
-        return self._dataset[item]
+        if self._field_instance.filter is not None and not isinstance(self._field_instance, IndexedStringField):
+            mask = self._field_instance.filter[:]
+            data = self._dataset[:][mask]  # as HDF5 does not support unordered mask
```

Comment:
Same idea here, want to get indices from mask rather than resolve all mask indices then select from result.